crates/assistant/src/assistant_settings.rs

```diff
@@ -12,14 +13,25 @@ pub enum OpenAiModel {
     Four,
     #[serde(rename = "gpt-4-1106-preview")]
     FourTurbo,
+    Custom,
 }
 
 impl OpenAiModel {
+    pub fn convert_enum(model_name: String) -> OpenAiModel {
+        match model_name.as_str() {
+            "gpt-3.5-turbo-0613" => OpenAiModel::ThreePointFiveTurbo,
+            "gpt-4-0613" => OpenAiModel::Four,
+            "gpt-4-1106-preview" => OpenAiModel::FourTurbo,
+            _ => OpenAiModel::Custom,
+        }
+    }
+
     pub fn full_name(&self) -> &'static str {
         match self {
             OpenAiModel::ThreePointFiveTurbo => "gpt-3.5-turbo-0613",
             OpenAiModel::Four => "gpt-4-0613",
             OpenAiModel::FourTurbo => "gpt-4-1106-preview",
+            OpenAiModel::Custom => "Custom model",
```

Comment:
The "model" here seems slightly redundant:

```suggestion
            OpenAiModel::Custom => "Custom",
```